ui/src/app/core/job-manager.service.ts

```diff
@@ -160,4 +160,24 @@ export class JobManagerService {
       .then(response => this.convertToAggregationJobsResponse(response.json()))
       .catch((e) => this.handleError(e));
   }
+
+  getTaskAttempts(id:string, task:string): Promise<JobAttemptsResponse> {
+    const apiUrl = this.configLoader.getEnvironmentConfigSynchronous()['apiUrl'];
+    return this.http.get(`${apiUrl}/jobs/${id}/${task}/attempts`,
+      new RequestOptions({headers: this.getHttpHeaders()}))
+      .toPromise()
+      .then(response => response.json())
+      .catch((e) => this.handleError(e));
+
+  }
+  //
+  getShardAttempts(id:string, task:string, index:number): Promise<JobAttemptsResponse> {
+    const apiUrl = this.configLoader.getEnvironmentConfigSynchronous()['apiUrl'];
+    return this.http.get(`${apiUrl}/jobs/${id}/${task}/${index}/attempts`,
```

Comment:
This and the function above appear the same except for this one line, seems like a DRYing opportunity. 🙂